### PR TITLE
Added end/withdraw all auctions button

### PR DESCRIPTION
**Pull request by Valentine1898**
Demo: 

https://github.com/user-attachments/assets/764af8a2-e251-40c8-9d11-808feee305a9

The limit of max 48 auctions in one transaction is set, but I still have the bug https://github.com/prax-wallet/web/issues/130 , reviewers, please let me know if you have the same.

### CODE REVIEW

File: apps/minifront/src/components/swap/auction-list/claim-all-button.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return (
+      <TooltipProvider>
+        <Tooltip>
+          <TooltipTrigger
+            onClick={() =>
+              void endAllAuctions(
+                // Chain has a transaction size limit, so we can add at most a batch of 48 auctions in a single transaction
+                // see https://github.com/penumbra-zone/web/issues/1166#issuecomment-2263550249
+                filterWithLimit(data, a => a.localSeqNum === 0n, 48),
+                // TODO Should use the index of the selected account after the account selector for the auction is implemented
```
Comment (Valentine1898) [inline apps/minifront/src/components/swap/auction-list/claim-all-button.tsx:36]: Currently, we can end/withdraw all only for Account 0 
see https://github.com/penumbra-zone/web/issues/1166#issuecomment-2265680885

I will open a separate issue for this to add an account selector to the auction list

File: apps/minifront/src/components/swap/auction-list/claim-all-button.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  source: AddressIndex;
+}
+
+// Assemble batch auctions for end or withdrawal.
+// All auctions in the batch will have the same 'AddressIndex'
+export const assembleAuctionBatch = (
+  auctions: AuctionInfo[],
+  filteredSeqNumber: bigint,
+  batchLimit: number,
+): AuctionsBatch => {
```
Comment (Valentine1898) [inline apps/minifront/src/components/swap/auction-list/claim-all-button.tsx:31]: This is a temporary solution that allows you to close all auctions on accounts with different batches
If a user has auctions on **account 0** and **account 1,** the first transaction will end all auctions on **account 0**, and the second transaction will end all auctions on **account 1**

File: apps/minifront/src/state/swap/dutch-auction/index.ts [hunk 3]
```diff
@@ -196,7 +202,31 @@ export const createDutchAuctionSlice = (): SliceCreator<DutchAuctionSlice> => (s
...
+      dutchAuctionEndActions: auctions.map(
+        au => new TransactionPlannerRequest_ActionDutchAuctionEnd({ auctionId: au.id }),
+      ),
+      source: addressIndex,
+    });
+    await planBuildBroadcast('dutchAuctionEnd', req);
+    get().swap.dutchAuction.auctionInfos.revalidate();
+  },
+  withdrawAllAuctions: async (auctions: AuctionInfo[], addressIndex: AddressIndex) => {
+    console.log(auctions);
```
Comment (jessepinho) [inline apps/minifront/src/state/swap/dutch-auction/index.ts:216]:
```suggestion
```

File: apps/minifront/src/components/swap/auction-list/claim-all-button.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                auctionBatch.auctions,
+                // TODO Should use the index of the selected account after the account selector for the auction is implemented
+                auctionBatch.source,
+              );
+            }}
+            aria-label='End all open auctions, with a limit of 48 auctions per transaction'
+          >
+            <div className='w-[85px] shrink-0'>
+              <Button size='sm' variant='secondary' className='w-full'>
+                {'End all'}
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/auction-list/claim-all-button.tsx:73]:
```suggestion
                End all
```

File: apps/minifront/src/components/swap/auction-list/claim-all-button.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                auctionBatch.auctions,
+                // TODO Should use the index of the selected account after the account selector for the auction is implemented
+                auctionBatch.source,
+              );
+            }}
+            aria-label='Withdraw all ended auctions, with a limit of 48 auctions per transaction'
+          >
+            <div className='w-[95px] shrink-0'>
+              <Button size='sm' variant='secondary' className='w-full'>
+                {'Withdraw all'}
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/auction-list/claim-all-button.tsx:104]:
```suggestion
                Withdraw all
```

**Review by jessepinho — APPROVED** [review]
The logic looks good to me, and could be merged as-is. However, there is a lot of repetition in the `ClaimAllButton` (which I'd also rename to `EndOrWithdrawAllButton`). If you could DRY that up, that'd be great — but feel free to merge this first to get this in ASAP.

**Review by hdevalence — APPROVED** [review]
Only watched UI video but looks good to me

**Pull request merged**